crates/sui/src/client_commands.rs

```diff
@@ -1251,6 +1267,36 @@ impl Display for SuiClientCommandResult {
                 }
                 writeln!(writer, "Showing {} results.", object_refs.len())?;
             }
+            SuiClientCommandResult::DynamicFieldQuery(df_refs) => {
+                writeln!(
+                    writer,
+                    "{0: ^67} | {1: ^15} | {2: ^63} | {3: ^42} | {4: ^10} | {5: ^44}",
+                    "Name", "Type", "Object Type", "Object Id", "Version", "Digest"
+                )?;
+                writeln!(writer, "{}", ["-"; 165].join(""))?;
+                loop{
```

Comment:
this loop is printing the same data over and over again if next_cursor is not None, until out of memory.

The loop to fetch the data pages should happen in line 517 instead and pass a Dec<DynamicFieldInfo> to here.